br/pkg/lightning/mydump/csv_parser.go

```diff
@@ -447,9 +448,17 @@ outside:
 
 func (parser *CSVParser) readQuotedField() error {
 	for {
+		prevPos := parser.pos
 		content, terminator, err := parser.readUntil(&parser.quoteByteSet)
-		err = parser.replaceEOF(err, errUnterminatedQuotedField)
 		if err != nil {
+			if errors.Cause(err) == io.EOF {
+				// return the position of quote to the caller.
+				// because we return an error here, the parser won't
+				// use the `pos` again, so it's safe to modify it here.
+				parser.pos = prevPos - 1
+				err = parser.replaceEOF(err, errUnterminatedQuotedField)
+			}
+			parser.Logger.Error("err content", zap.ByteString("content", content))
```

Comment:
```suggestion
```

Reply:
Already logged in `logSyntaxError`.

Reply:
No, it won't be logged in `logSyntaError`:
```
[2023/01/05 22:32:09.314 +08:00] [ERROR] [parser.go:208] ["syntax error"] [pos=30] [content=]
[2023/01/05 22:32:09.314 +08:00] [ERROR] [csv_parser.go:461] ["err content"] [content=",7,8\n"]
```
The reason is that `parser.buf` has already been set to nil and only points to the content if the parser doesn't meet error.
```golang
for {
		buf = append(buf, parser.buf...)
		parser.buf = nil
		if err := parser.readBlock(); err != nil || len(parser.buf) == 0 {
			if err == nil {
				err = io.EOF
			}
			parser.pos += int64(len(buf))
			return buf, 0, errors.Trace(err)
		}
                 index := IndexAnyByte(parser.buf, chars)
		if index >= 0 {
			buf = append(buf, parser.buf[:index]...)
			parser.buf = parser.buf[index:]
			parser.pos += int64(len(buf))
			return buf, parser.buf[0], nil
		}
```

Reply:
Could we let `logSyntaError` print the content instead of a new log which looks a little confusing?

Reply:
What about set `parser.buf = buf` if we meet an EOF error?

Reply:
> What about set `parser.buf = buf` if we meet an EOF error?

LGTM

Reply:
66f141f

Reply:
5a55279